### DISCUSSION TITLE
Text Wrap Around Image

### DISCUSSION THREAD

**Discussion by cab-1729** [ImageMagick/ImageMagick#7457] [General] [answered]
How do I add labels that wrap around an image, or a leave a rectangular gap for me to place an image, without breaking up the text on my own? Suppose I have my text in a ```pango``` file, and I have ```-composite```d an Image onto a blank or monochrome canvas. How do I add the text to the canvas to wrap around the image. Even wrapping around the image on one side fulfills my use case.

I aim to do this with MagickWand but showing me the way to do it with the cli would help me understand.

**Comment by snibgo** [comment]
I don't understand what you mean by "labels that wrap around an image". Can you supply example images?

**Reply by cab-1729** [reply]
![image_wrap_align](https://github.com/ImageMagick/ImageMagick/assets/60106638/9553149e-e048-4f05-ab7f-d074950a120a)
<img width="1042" alt="Co4Bp" src="https://github.com/ImageMagick/ImageMagick/assets/60106638/266e67ce-a702-489f-9952-a5616d3e0c00">

**Comment by fmw42** [comment]
There is nothing in Imagemagick that will automatically wrap text around images or leave spaces.  Perhaps Pango can do that, but I would not know if that is even possible.

In Imagemagick, you would have to create multiple sections of text images (using caption:) designed to fit each different text area that you want on every side of the image and composite them all (text sections and image) together.

Perhaps one of the other users has a better idea.

**Comment by snibgo** [comment, marked as answer]
ImageMagick doesn't contain code for rendering text within a specified area.

SVG does have that facility, and Inkscape can render it, and ImageMagick can use Inkscape as a delegate. See, for example, https://www.w3.org/TR/SVG2/text.html#TextShapeInside .
